### PR TITLE
Clean up keys after done with signing vault

### PR DESCRIPTION
**Pull request by junzhli**
Hi,

I notice the attached sign.command that it leaves Keys folder in EFI/OC/
To my understanding, those keys no longer need during the boot process. Its public key was compiled into opencore.efi
I think it would be safer to remove those keys after done with this script
Also, I put the folder Keys under temp folder "/tmp" to prevent create_vault from processing them.

### CODE REVIEW

File: Utilities/CreateVault/sign.command [hunk 1]
```diff
@@ -77,5 +77,10 @@ fi
 
 /bin/dd of="${OCBin}" if="${PubKey}" bs=1 seek="${off}" count=528 conv=notrunc || abort "Failed to bin-patch ${OCBin}"
 
+echo "Cleaning up keys"
+if [ -d "${KeyPath}" ]; then
+  rm -rf "${KeyPath}" | abort "Failed to clean up keys ${KeyPath}!"
```
Comment (vit9696) [inline Utilities/CreateVault/sign.command:82]: It is not `|` but `||`.

File: Utilities/CreateVault/sign.command [hunk 2]
```diff
@@ -77,5 +77,10 @@ fi
 
 /bin/dd of="${OCBin}" if="${PubKey}" bs=1 seek="${off}" count=528 conv=notrunc || abort "Failed to bin-patch ${OCBin}"
 
+echo "Cleaning up keys"
+if [ -d "${KeyPath}" ]; then
```
Comment (vit9696) [inline Utilities/CreateVault/sign.command:81]: `-d` is not necessary here, as you always delete.

**Comment by vit9696** [comment]
The implementation is not correct for two reasons:

1. `/tmp/Keys` directory may exist due to other software. Please make it `/tmp/Keys-$(/usr/bin/uuidgen)` to make sure it is unique. Furthermore directory existence prior to creation should now cause a fatal error, as `$(/usr/bin/uuidgen)` is pretty much guaranteed to produce a unique value. Also make sure that `/usr/bin/uuidgen` exists prior to executing it in the top of the script like other tools are.
2. Remove the directory not only in the normal workflow but also in abort to ensure that all the keys are dead regardless of script success or failure.

I also left a couple of notes inline.

**Comment by junzhli** [comment]
Hi @vit9696 ,
Thanks for the review. I push another revised version!

File: Utilities/CreateVault/sign.command [hunk 3]
```diff
@@ -77,5 +84,7 @@ fi
 
 /bin/dd of="${OCBin}" if="${PubKey}" bs=1 seek="${off}" count=528 conv=notrunc || abort "Failed to bin-patch ${OCBin}"
 
+
```
Comment (vit9696) [inline Utilities/CreateVault/sign.command:87]: Why is that necessary?

File: Utilities/CreateVault/sign.command [hunk 4]
```diff
@@ -45,6 +50,8 @@ if [ ! -x ./RsaTool ] || [ ! -x ./create_vault.sh ]; then
   fi
 fi
 
+trap cleanup EXIT ERR INT TERM
```
Comment (vit9696) [inline Utilities/CreateVault/sign.command:53]: Do you believe it is better to directly calling `cleanup` in e.g. `abort`? I personally do not mind.

**Comment by vit9696** [comment]
You may want to add an exception for that particular line with trap as currently CI fails.

**Comment by junzhli** [comment]
@vit9696,
I remove the unnecessary blank line and unsupported signals for adapting to POSIX-compilant
My concern about putting cleanup() in abort() function is that all of code above line 53 invoking abort() won't work properly as var KeyPath is not ready at that moment. Also it provides guarantee that cleanup gets called when users terminate the script by pressing Ctrl+C. Thanks!
